src/ui/flutter_app/lib/services/storage/storage_migration.dart

```diff
@@ -0,0 +1,190 @@
+class _DatabaseMigrator {
+  const _DatabaseMigrator._();
+
+  static const _tag = "[Database Migration]";
+
+  /// The newest DB version.
+  static const _newVersion = 2;
+
+  /// The current DB version.
+  /// It will get initialized by [migrate] with the saved value.
+  static late final int _currentVersion;
+
+  /// The list of migrations
+  static const _migrations = [
+    _migrateToHive,
+    _migrateFrom1,
+  ];
+
+  /// Database box reference
+  static late Box _databaseBox;
+
+  /// key at which the [_databaseBox] will be saved
+  static const String _databaseBoxName = "database";
+
+  /// key at which the database info will be saved in the [_databaseBox]
+  static const String _versionKey = "version";
+
+  /// Migrates the database version to the [_newVersion].
+  ///
+  /// When [_currentVersion] is smaller than [_newVersion] all migrations between will be run.
+  /// Migrations are called once and only once thus they should complete fast and everything should be awaited.
+  static Future<void> migrate() async {
+    assert(_migrations.length == _newVersion);
+
+    _databaseBox = await Hive.openBox(_databaseBoxName);
+
+    _currentVersion = _databaseBox.get(
+      _versionKey,
+      defaultValue: (!LocalDatabaseService.preferences.usesHiveDB) ? 0 : 1,
+    ) as int;
+
+    for (int i = _currentVersion; i < _newVersion; i++) {
+      await _migrations[i].call();
+    }
+
+    await _databaseBox.put(_versionKey, _newVersion);
+    _databaseBox.close();
+  }
+
+  /// Migration 0 - KV to Hive
+  ///
+  /// - Calls the [DatabaseV1.migrateDB] to migrate from KV storage.
+  static Future<void> _migrateToHive() async {
+    assert(_currentVersion <= 0);
+
+    await DatabaseV1.migrateDB();
+    debugPrint("$_tag migrated from KV");
+  }
+
+  /// Migration 1 - Sanmill version 1.1.38+2196 - 2021-11-09
+  ///
+  /// - **Algorithm to enum:** Migrates [LocalDatabaseService.preferences] to use the new Algorithm enum instead of an int representation.
+  /// - **Drawer background color:** Migrates [LocalDatabaseService.colorSettings] to merge the drawerBackgroundColor and drawerColor.
+  /// This reflects the deprecation of drawerBackgroundColor.
+  static Future<void> _migrateFrom1() async {
+    assert(_currentVersion <= 1);
+
+    final _prefs = LocalDatabaseService.preferences;
+    LocalDatabaseService.preferences = _prefs.copyWith(
+      algorithm: Algorithms.values[_prefs.oldAlgorithm],
+    );
+
+    final _colorSettings = LocalDatabaseService.colorSettings;
+    LocalDatabaseService.colorSettings = _colorSettings.copyWith(
+      drawerColor: Color.lerp(
+        _colorSettings.drawerColor,
+        _colorSettings.drawerBackgroundColor,
+        0.5,
+      )?.withAlpha(0xFF),
+    );
+
+    debugPrint("$_tag migrated from v1");
+  }
+}
+
+/// Database KV Migrator
+///
+/// This class provides helper methods to migrate from the old KV storage to the new hiveDB.
+class DatabaseV1 {
```

Comment:
OK. The Sanmill's next version will rise to v2.0.0.

Reply:
still have a lot of changes in the pipeline though 😉

Also the database version is not related to the app version